packages/dtslint/src/checks.ts

```diff
@@ -24,14 +24,34 @@ export type CompilerOptionsRaw = {
     ? string | number | undefined
     : CompilerOptions[K];
 };
+interface Tsconfig {
+  compilerOptions: CompilerOptionsRaw;
+  files?: string[];
+  include?: string[];
+  exclude?: string[];
+}
 
-export function checkTsconfig(dirPath: string, options: CompilerOptionsRaw): string[] {
+export function checkTsconfig(dirPath: string, config: Tsconfig): string[] {
   const errors = [];
   const mustHave = {
     noEmit: true,
     forceConsistentCasingInFileNames: true,
     types: [],
   };
+  const options = config.compilerOptions;
+  if ("include" in config) {
+    errors.push('Use "files" instead of "include".');
+  }
+  if ("exclude" in config) {
+    errors.push('Use "files" instead of "exclude".');
+  }
+  if (!config.files) {
+    errors.push('Must specify "files".');
+  } else if (!config.files.some((f) => f.endsWith(".d.ts"))) {
```

Comment:
I was going to suggest that you use `isDeclarationPath` here, which is _technically_ the right answer, but I actually think that we should be checking to verify that `index.d.ts` in particular is present (with a todo for `isDeclaraionPath`).

The reason is that there are two eslint rules which only run on the "main" file, which is hardcoded `index.d.ts`, and they both rely on being able to ask eslint for the Program based on that file (so, `index.d.ts` must be in the Program).